File: Decision Tree Regression/decision_tree_regression.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1:2].values
y = dataset.iloc[:,2].values

from sklearn.tree import DecisionTreeRegressor

regressor = DecisionTreeRegressor()
regressor.fit(X,y)

y_pred = regressor.predict(6.5)

X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid),1))
plt.scatter(X,y,color='red')
plt.plot(X_grid,regressor.predict(X_grid),color='blue')
plt.title('Truth or bluff(Decision Tree Regression)')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.show()

File: Regression models/Decision Tree Regression/decision_tree_regression.R
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

#install.packages('rpart')
library(rpart)
regressor = rpart(formula = Salary ~ .,
                  data = dataset,
                  control = rpart.control(minsplit = 1))

y_pred = predict(regressor, data.frame(Level=6.5))

library(ggplot2)
X_grid = seq(min(dataset$Level),max(dataset$Level),0.01)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = 'red') +
  geom_line(aes(x = X_grid , y = predict(regressor,newdata = data.frame(Level = X_grid))),
             color = 'blue') +
  ggtitle('Truth or bluff(Decision Tree Regression') +
  xlab('Levels') +
  ylab('Salary')